### DISCUSSION TITLE
On_boost option in a listener?

### DISCUSSION THREAD

**Discussion by Timmiegamer** [nextcord/nextcord#761] [Q&A] [answered]
Maybe it can be usefull if there is an option like on_server_boost so you can add that to a listener and and make custom boosting messages for you community's.

**Comment by DenverCoder1** [comment, marked as answer]
It's not an event sent by Discord, but there are a few ways to do it depending on what you want to achieve:

1. You can use `on_member_update` and check for changes to `premium_since` between before and after
2. You can use `on_guild_update` and check for differences in `premium_subscribers` / `premium_subscriber_count` (this won't tell you which user boosted if they boost a second time, though)
3. Check for system messages with `message.type` being one of the `MessageType.premium_guild_*` types.
